app/assets/javascripts/Components/__tests__/marks_panel.test.jsx

```diff
@@ -0,0 +1,425 @@
+import {shallow} from "enzyme";
+
+import {
+  MarksPanel,
+  CheckboxCriterionInput,
+  FlexibleCriterionInput,
+  RubricCriterionInput,
+} from "../Result/marks_panel";
+
+const convertToKebabCase = {
+  CheckboxCriterion: "checkbox_criterion",
+  FlexibleCriterion: "flexible_criterion",
+  RubricCriterion: "rubric_criterion",
+};
+
+describe("MarksPanel", () => {
+  let basicProps;
+  const getWrapper = props => {
+    return shallow(<MarksPanel {...props} />);
+  };
+  beforeEach(() => {
+    basicProps = {
+      max_mark: 100,
+      released_to_students: false,
+      old_marks: {},
+      marks: [],
+      levels: [],
+      assigned_criteria: [],
+      annotations: [],
+      updateMark: jest.fn(),
+      revertToAutomaticDeductions: jest.fn(),
+      findDeductiveAnnotation: jest.fn(),
+    };
+  });
+  it("displays no criterion if marks is an empty list", () => {
+    const wrapper = getWrapper();
+    expect(wrapper.find("marks-list").length).toBe(0);
+  });
+  it("displays a criterion of the right type when given an array of marks", () => {
+    basicProps.marks.push({
+      criterion_type: "CheckboxCriterion",
+      id: 67,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[67] = [];
+    basicProps.marks.push({
+      criterion_type: "FlexibleCriterion",
+      id: 66,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[66] = [];
+    basicProps.marks.push({
+      criterion_type: "RubricCriterion",
+      id: 65,
+      mark: 0,
+      description: "testing",
+    });
+    basicProps.old_marks[65] = [];
+    const wrapper = getWrapper(basicProps);
+    basicProps.marks.forEach(mark => {
+      expect(wrapper.find(`#${convertToKebabCase[mark.criterion_type]}_${mark.id}`)).toBeTruthy();
+    });
+  });
+});
+
+describe("CheckboxCriterionInput", () => {
+  let wrapper, basicProps;
+  beforeEach(() => {
+    basicProps = {
+      id: 67,
+      mark: 0,
+      max_mark: 1,
+      expanded: false,
+      unassigned: false,
+      released_to_students: false,
+      oldMark: {},
+      description: " ",
+      updateMark: jest.fn(),
+      destroyMark: jest.fn(),
+      toggleExpanded: jest.fn(),
+    };
+    wrapper = shallow(<CheckboxCriterionInput {...basicProps} />);
+  });
+  it("should toggle expand and contract upon clicking the expand/contract button", () => {
+    basicProps.toggleExpanded.mockImplementation(() => {
```

Comment:
It seems to me that this can be simplified by putting this function inside the call to `jest.fn()` above. (I'm fine with it being defined on every test, even if the function is only called in some of the tests).

Reply:
Each time I mock toggleExpanded it is in a different describe block and refers to a different basicProps. This results in it only being mocked once per test suite. I don't necessarily think this would simplify it or maybe I am misunderstanding your suggestion. For now I will leave it as-is until i get clarification

Reply:
@WeissSa What I suggested will simplify the test cases themselves by moving the setup code into a setup function (`beforeEach`). You're correct that currently these functions are only mocked in one or two tests, but this may change in the future.

Reply:
Ah okay! I just wanted to make sure we are on the same page then. I'll move the location of both next time I get a chance to.